### DISCUSSION TITLE
Direct link to the image in Immich

### DISCUSSION THREAD

**Discussion by redge76** [damongolding/immich-kiosk#173] [Feature request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Hi,

Would it be possible to add a link to the picture in immich ?
I loop though all my pictures and sometime I need to access the picture to know where is was taken.
This link could be where is the icons for "left" "play" "right" are.

**Comment by damongolding** [comment]
So the link would open the Immich UI on said image?

My first thought would be to add a link button/icon on the image when paused. Possibly better for UX but also doesn't bloat the menu.

**Reply by redge76** [reply]
That's a possibility. A small "i" on top would be a possibility but it would "bloat" the rest of the screen...
Pixette has the "i" in the menu and it's pretty clean.
![image](https://github.com/user-attachments/assets/e3a1f25b-c983-4b19-8ea0-05f90f7d6394)

**Reply by damongolding** [reply]
The reason I suggested a button on the image itself is for when users are using the splitview layout which has 2 images on the screen. 

Does the "i" icon display an overlay on the image? That could also work.

**Reply by redge76** [reply]
oh Yes good point. Then a small "i" icon would do. 

Or 2 "i" icon in the menu. One on the left and one on the right with small arrows. 
But this would mean that the menu would be different when there is one or two pictures.
(not sure this would be an issue...)
![image](https://github.com/user-attachments/assets/f50be25c-d748-4a2b-81bb-3de0e086d4f9)

**Reply by damongolding** [reply]
I'm leaning towards an "i" in the menu and once pressed an overlay would show over the image(s) displaying additional info, including a link to the image in Immich. 

The tricky bit is the interaction with the hitboxs that cover the whole screen.

**Reply by redge76** [reply]
Hi, 

The latest interface is great. The QR code is an excellent idea but my immich URL for external access is protected behind a reverse proxy with some google authentication (Cloudflare Access). (immich-kiosk is protected too...) so the generated image url is not usable.
The KIOSK_IMMICH_URL is an unprotected endpoint only available for the docker compose services.

Would it be possible to have an "KIOSK_IMMICH_EXTERNAL_URL" that is used for all generated image URLs and Qrcode  ?

Thanks

**Reply by damongolding** [reply]
I meant to add this. Thank you for reminding me!

**Reply by redge76** [reply]
and if your are in this park of you code, can you make a button to the location url ?
http://maps.google.com/?q=[LOCATION OF PICTURE]
This button would be just bellow the "view image in immich" button

Also can you repeat the images name, date , location in the information above the "view" button. 
Currently these information are impossible to copy/paste from the image. (before you click in "i" )

**Reply by redge76** [reply]
And center everything
![image](https://github.com/user-attachments/assets/6ba19ae9-71a4-4f44-ac7c-c119ba20dce9)

**Reply by redge76** [reply]
Something like (no need to blur whole screen... but I don't know how this would display on a tablet or phone)
![image](https://github.com/user-attachments/assets/41399258-f97b-4b77-8c34-e864222e065f)

**Reply by damongolding** [reply]
Thanks for the suggestions!

**Comment by damongolding** [comment]
I've added the meta data to the overlay and also made the location a clickable link to google maps
